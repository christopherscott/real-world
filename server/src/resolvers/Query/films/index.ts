import { Context } from '../../../utils';
import { Film } from '../../../generated/types';

export const films = (parent, args, context: Context, info) => {
  // Order by multiple fields is not supported yet
  // https://github.com/prismagraphql/prisma/issues/62
  return context.db.query.films({ orderBy: 'rating_DESC' }, info);
};

export const film = (
  parent,
  { id }: { id: Film['id'] },
  context: Context,
  info
) => {
  return context.db.query.film({ where: { id } }, info);
};
