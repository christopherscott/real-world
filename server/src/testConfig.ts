import { GraphQLClient } from 'graphql-request';
import { Prisma } from './generated/prisma';

const database = new Prisma({
  endpoint: 'http://localhost:4466',
  secret: 'mysecret123',
});

export const client = new GraphQLClient(`http://localhost:4000`);

export const db = {
  clean: async () => {
    await database.mutation.deleteManyGenres({});
    await database.mutation.deleteManyComments({});
    await database.mutation.deleteManyPersons({});
    await database.mutation.deleteManyFilms({});
  },
  ...database,
};
