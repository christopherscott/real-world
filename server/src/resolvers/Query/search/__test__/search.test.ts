import { map } from 'ramda';
import { db, client } from '../../../../testConfig';
import films from './fixtures/films';
import user from './fixtures/user';

afterEach(async () => {
  await db.clean();
});

test('should return films and comment matched by search word', async () => {
  const filmIds = await Promise.all(
    map(data => db.mutation.createFilm({ data }, '{ id }'), films)
  );

  const { id: userId } = await db.mutation.createUser({ data: user }, '{ id }');

  await db.mutation.createComment({
    data: {
      user: { connect: { id: userId } },
      film: { connect: { id: filmIds[0].id } },
      text: 'Foo',
    },
  });

  const query = `
    query {
      search(search: "Foo") {
        ... on Film {
          title
          plot
          rating
        }
        ... on Comment {
          text
        }
      }
    }
  `;

  const responseData = await client.request(query);

  expect(responseData).toMatchSnapshot();
});
