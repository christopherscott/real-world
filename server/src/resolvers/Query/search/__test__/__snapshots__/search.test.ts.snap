// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return films and comment matched by search word 1`] = `
Object {
  "search": Array [
    Object {
      "plot": "Foo Two imprisoned men bond over a number of years, finding solace and eventual redemption through acts of common decency.",
      "rating": 9.3,
      "title": "The Shawshank Redemption",
    },
    Object {
      "text": "Foo",
    },
  ],
}
`;
